### PR TITLE
Feature/generate user data

### PR DESCRIPTION
**Pull request by ostrya**
Since #121   did not see any update for a long time, I decided to implement the desired feature myself.

### CODE REVIEW

File: mock/src/main/java/com/tngtech/keycloakmock/api/TokenConfig.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
  * <p>Example usage:
  *
  * <pre>{@code
- * TokenConfig config = TokenConfig.aTokenConfig().withSubject("subject).build();
```
Comment (space-hawk) [inline mock/src/main/java/com/tngtech/keycloakmock/api/TokenConfig.java:30]: was this not broken?

File: mock/src/main/java/com/tngtech/keycloakmock/impl/TokenGenerator.java [hunk 2]
```diff
@@ -56,12 +55,41 @@ public String getToken(
             .claim("scope", tokenConfig.getScope())
             .claim("typ", "Bearer")
             .claim("azp", tokenConfig.getAuthorizedParty());
+    Optional<UserData> generatedUserData;
```
Comment (space-hawk) [inline mock/src/main/java/com/tngtech/keycloakmock/impl/TokenGenerator.java:58]: Could use ternary

**Review by space-hawk — APPROVED** [review]

**Review by space-hawk — APPROVED** [review]

File: mock/src/main/java/com/tngtech/keycloakmock/api/TokenConfig.java (hunk 1 above)
Comment (ostrya) [reply mock/src/main/java/com/tngtech/keycloakmock/api/TokenConfig.java:30]: I'm not quite sure what you mean. If you're talking about the missing quotation mark: yes, that was broken.

**Review by ripssi — APPROVED** [review]
Had to change account to submit review. See comments by space-hawk ...

File: mock/src/main/java/com/tngtech/keycloakmock/impl/TokenGenerator.java (hunk 2 above)
Comment (ostrya) [reply mock/src/main/java/com/tngtech/keycloakmock/impl/TokenGenerator.java:58]: Hm, yes. But it's not significantly shorter, and the google formatting looks weird:

```
    Optional<UserData> generatedUserData =
        tokenConfig.isGenerateUserDataFromSubject() ? Optional.of(
            UserData.fromUsernameAndHostname(
                tokenConfig.getSubject(), requestConfiguration.getHostname())) : Optional.empty();
```

**Pull request merged**